### ISSUE TITLE
Source generator warning duplication and missing link and description

### ISSUE THREAD

**Issue by kzu** [dotnet/roslyn#69826]
**Version Used**: 

* Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview Version 17.8.0 Preview 1.0

**Steps to Reproduce**:

1. Clone/download https://github.com/kzu/SourceGeneratorWarning/
2. Build once: a warning should appear, with a link, message and description
3. Rebuild the Tests project. A duplicate warning missing the right link (takes you instead to a generic VS F1 help page) and missing the description is show:

![image](https://github.com/dotnet/roslyn/assets/169707/903e4504-d38b-45a3-9c4b-02803bd5476e)

**Expected Behavior**:

Only one warning is ever shown, and it is consistent, showing the description and link as provided by the diagnostic descriptor.

**Actual Behavior**:

Duplicate warning, missing link and description.

**Comment by jcouv** [comment]
Could you check what happens if you change the "Build + Intellisense" drop-down to either "Build" or "Intellisense"? It feels like we may not be properly recognizing duplicates from those two sources of diagnostics...

**Comment by kzu** [comment]
Indeed. The missing data item is the one coming from Build, not Intellisense.

![image](https://github.com/dotnet/roslyn/assets/169707/41a77b33-62d0-43e1-a857-8f98bd5a3827)

![image](https://github.com/dotnet/roslyn/assets/169707/881afa68-9b94-4acb-8cf1-01bb731bc781)
 
![image](https://github.com/dotnet/roslyn/assets/169707/d3c73ca9-d536-44e2-a32e-5445eae208b6)

**Comment by jcouv** [comment]
Tagging @mavasani to advise. Does this look like an IDE or a compiler issue?

**Comment by mavasani** [comment]
@jcouv Sorry I missed this in my inbox. This is definitely an IDE issue. We have special code in the IDE to associate the descriptor properties, such as helplink, to build reported diagnostics. This code only handles analyzer reported diagnostics and descriptors, not from source generators.

**Comment by kzu** [comment]
Is this still on track for 17.10? 🙏 

Thanks!

**Comment by kzu** [comment]
This is still happening in 17.12 (notice the missing hyperlink now):

![image](https://github.com/user-attachments/assets/2662a5ae-bc0b-485c-bf86-a348489f2e58)

Microsoft Visual Studio Enterprise 2022
Version 17.12.0 Preview 1.0
VisualStudio.17.Preview/17.12.0-pre.1.0+35209.166
C# Tools   4.12.0-1.24379.11+cf82d399c36008e7936d545cde24141f8d3790fa

**Comment by dennis-garavsky** [comment]
We hit the same issue at DevExpress for our "shift-left" code diagnostics like https://docs.devexpress.com/eXpressAppFramework/403389/debugging-testing-and-error-handling/code-diagnostics

Currently, two messages are shown in the Error List window instead of 1. The 1st message is cut off and incomplete, the 2nd is fine.
![Image](https://github.com/user-attachments/assets/b631b173-3fad-4093-ab99-eed862e469b8)

If we try to apply a workaround from https://github.com/dotnet/roslyn/issues/48083#issuecomment-1135227677 , then the second message is gone, but the message text is cut off/incomplete:

![Image](https://github.com/user-attachments/assets/5b681317-7b4f-46bb-915f-58d9dee1a7bb)

Please suggest a reliable solution or a fix. Thank you.
